### PR TITLE
Clean codes in EdmLibExtensions for coverage

### PR DESCRIPTION
**Pull request by xuzhg**
### Description

Related to test coverage.

So far, it has the following code coverage:

image

### Checklist (Uncheck if it is not completed)

- [ x ] Test cases added
- [ x ] Build and test with one-click build and test script passed

### CODE REVIEW

File: src/Microsoft.OData.Edm/ExtensionMethods/ExtensionMethods.cs [hunk 1]
```diff
@@ -1641,6 +1641,35 @@ public static bool IsOpen(this IEdmStructuredTypeReference type)
...
+            if (structuredType != null)
+            {
+                return structuredType.IsOpen;
+            }
+
+            // If its a collection, return whether its element type is open.
+            // This is because when processing a navigation property, the target type
+            // may be a collection type even though a key expression has been applied.
+            // This will be cleaned up in a subsequent change.
+            // TODO: when SingleResult is removed from the semantic path parser, change this to return false.
```
Comment (AlanWong-MS) [inline src/Microsoft.OData.Edm/ExtensionMethods/ExtensionMethods.cs]: Do we have a way to track this TODO? Should we open a GitHub issue and reference the number here?

File: src/Microsoft.OData.Core/UriParser/Binders/EndPathBinder.cs [hunk 2]
```diff
@@ -111,7 +111,7 @@ internal static SingleValueNode CreateParentFromImplicitRangeVariable(BindingSta
         internal SingleValueOpenPropertyAccessNode GeneratePropertyAccessQueryForOpenType(EndPathToken endPathToken, SingleValueNode parentNode)
         {
             if (parentNode.TypeReference == null ||
-                parentNode.TypeReference.Definition.IsOpenType() ||
```
Comment (AlanWong-MS) [inline src/Microsoft.OData.Core/UriParser/Binders/EndPathBinder.cs]: What's the reasoning for moving from .IsOpenType() to .IsOpen()?

Comment (xuzhg) [inline src/Microsoft.OData.Core/UriParser/Binders/EndPathBinder.cs]: EdmLib has `IsOpen()` extension method defined ,  `IsOpenType()` was similar function defined in ODL and I removed it.

File: src/Microsoft.OData.Edm/ExtensionMethods/ExtensionMethods.cs (hunk 1 above)
Comment (xuzhg) [inline src/Microsoft.OData.Edm/ExtensionMethods/ExtensionMethods.cs]: Oh, I think we should remove this "TODO". I just copied from the ODL lib.

**Comment by AlanWong-MS** [comment]
:shipit:

**Comment by xuzhg** [comment]
Merged at: 5ece8b56164f44c4b09101362953e0a76b6591d0

**Pull request closed without merging**